src/cmd/ksh93/bltins/whence.c

```diff
@@ -192,7 +192,7 @@ static int whence(Shell_t *shp,char **argv, register int flags)
 		}
 		/* built-ins and functions next */
 	bltins:
-		if(!(flags&F_FLAG) && (np = nv_bfsearch(name, shp->fun_tree, &nq, &notused)) && !is_abuiltin(np))
+		if(!(flags&F_FLAG) && (np = nv_bfsearch(name, shp->fun_tree, &nq, &notused)) && !is_abuiltin(np) && nv_isattr(np, NV_FUNCTION))
```

Comment:
I *think* the `!is_abuiltin(np) && nv_isattr(np, NV_FUNCTION)` can be replaced by `is_afunction(np)`. Not sure why that wasn't used to begin with.

Reply:
I've just tried the suggested change, but it introduces regression test failures:
```
$ bin/shtests -p builtins path subshell
#### Regression-testing /home/johno/GitRepos/KornShell/ksh/arch/linux.i386-64/bin/ksh ####
test builtins begins at 2021-04-24+11:54:18
	builtins.sh[240]: whence -v test not a builtin
test builtins failed at 2021-04-24+11:54:26 with exit code 1 [ 234 tests 1 error ]
test path begins at 2021-04-24+11:54:26
	path.sh[584]: 'whence -a' does not work correctly with tracked aliases (expected $'chmod is a shell builtin\nchmod is a tracked alias for /bin/chmod', got $'chmod is a function\nchmod is a shell builtin\nchmod is a tracked alias for /bin/chmod')
	path.sh[590]: 'whence -a': incorrect output (expected $'printf is a shell builtin\nprintf is /bin/printf', got $'printf is a function\nprintf is a shell builtin\nprintf is /bin/printf')
	path.sh[598]: 'whence -a': incorrect output for function+builtin (expected $'printf is a function\nprintf is a shell builtin\nprintf is /bin/printf', got $'printf is a shell builtin\nprintf is /bin/printf')
	path.sh[604]: 'whence -a': incorrect output for autoload+builtin (expected $'printf is an undefined function\nprintf is a shell builtin\nprintf is /bin/printf', got $'printf is a shell builtin\nprintf is /bin/printf')
	path.sh[677]: failure in reporting or running autoloadable functions -- diff follows:
	| --- /dev/fd/3	2021-04-24 11:54:28.576835642 -0700
	| +++ /dev/fd/4	2021-04-24 11:54:28.576835642 -0700
	| @@ -1,7 +1,5 @@
	| -chmod is an undefined function (autoload from /tmp/ksh93.shtests.28356.26422/path.C/whencefun/chmod)
	| -chmod is /bin/chmod
	| +chmod is a tracked alias for /bin/chmod
	|  whence_FPATH_test is an undefined function (autoload from /tmp/ksh93.shtests.28356.26422/path.C/whencefun/whence_FPATH_test)
	| -whence_autoload_test is an undefined function (autoload from /tmp/ksh93.shtests.28356.26422/path.C/whencefun/whence_autoload_test)
	|  ls is a tracked alias for /bin/ls
	|  ls is an undefined function (autoload from /tmp/ksh93.shtests.28356.26422/path.C/whencefun/ls)
	|  cp is a tracked alias for /bin/cp
test path failed at 2021-04-24+11:54:28 with exit code 5 [ 138 tests 5 errors ]
test subshell begins at 2021-04-24+11:54:28
	subshell.sh[709]: type/whence -v finds function in virtual subshell after it's unset with 'unset -f' (expected Success, got 'notafunc is an undefined function')
test subshell failed at 2021-04-24+11:54:34 with exit code 1 [ 124 tests 1 error ]
Total errors: 7
```

Reply:
Actually, ignore my previous post. I made a typo (`!is_afunction` instead of `is_afunction`) that caused regression test failures. After applying the change correctly, it works fine now. Now applied to the pull request.